docs/Versions.asciidoc

```diff
@@ -58,6 +58,16 @@ endif::[]
 :javadoc-license: {rest-high-level-client-javadoc}/org/elasticsearch/protocol/xpack/license
 :javadoc-watcher: {rest-high-level-client-javadoc}/org/elasticsearch/protocol/xpack/watcher
 
+///////
+Permanently unreleased branches (master, n.X)
+///////
+ifeval::["{source_branch}"=="master"]
+:permanently-unreleased-branch:
+endif::[]
+ifeval::["{source_branch}"=="{major-version}"]
+:permanently-unreleased-branch:
+endif::[]
```

Comment:
This is more messy than preferred, but Asciidoctor does not currently support complex conditions:
https://github.com/asciidoctor/asciidoctor/issues/1786